Assets/Scripts/Utilities/Logger.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using UnityEngine;
+
+public static class Logger
+{
+    public enum Level
+    {
+        Exception,
+        Assertion,
+        Error,
+        Warning,
+        Info,
+        Verbose
+    }
+
+    // TODO: Not used yet but it's the logical next step.
+    public enum Channel
+    {
+        Default
+    }
+
+    static Level minimumLevel = Level.Info;
+
+    public static void LogException(Exception exception)
+    {
+        // Exceptions are special as they support neither formatting nor message
+        if (minimumLevel <= Level.Exception)
+        {
+            Debug.LogException(exception);
+        }
+    }
+
+    public static void LogAssertion(string message)
+    {
+        Log(Level.Assertion, message);
+    }
+
+    public static void LogAssertionFormat(string format, params object[] args)
+    {
+        Log(Level.Assertion, format, args);
+    }
+
+    public static void LogError(string message)
+    {
+        Log(Level.Error, message);
+    }
+
+    public static void LogErrorFormat(string format, params object[] args)
+    {
+        Log(Level.Error, format, args);
+    }
+
+    public static void LogWarning(string message)
+    {
+        Log(Level.Warning, message);
+    }
+
+    public static void LogWarningFormat(string format, params object[] args)
+    {
+        Log(Level.Warning, format, args);
+    }
+
+    public static void LogInfo(string message)
+    {
+        Log(Level.Info, message);
+    }
+
+    public static void LogInfoFormat(string format, params object[] args)
+    {
+        Log(Level.Info, format, args);
+    }
+
+    public static void LogVerbose(string message)
```

Comment:
Add [ConditionalAttribute("DEBUG")] ?

So that all the LogVerbose functions don't even make it to a release build ? (Not sure if supported by Unity)

http://stackoverflow.com/questions/5080477/debug-only-code-that-should-run-only-when-turned-on